File: test/struct.js
var MyStruct = function(values) {
  this.k = values.k;
  this.k2 = values.k2;
};
MyStruct.$rtj_new = function $rtj_new(v) {
  return new MyStruct({
    k: v,
    k2: v * 2,
  });
};
function main() {
  var v = MyStruct.$rtj_new(1);
  console.log("" + (v . k) + "");
}

File: test/struct.rs
struct MyStruct {
  k: i64,
  k2: i64,
}

impl MyStruct {
  fn new(v: i64) -> Self {
    MyStruct {
      k: v,
      k2: v * 2,
    }
  }
}

fn main() {
  let v = MyStruct::new(1);
  println!("{}", v.k);
}
